File: 프로그래머스/lv2/42747. H－Index/H－Index.py
def solution(citations):
    answer = 0
    for i in range(len(citations) + 1):
        num1 = [a for a in citations if a >= i]
        num2 = [b for b in citations if b <= i]
        if len(num1) >= i and len(num2) <= i:
            answer = i
    return answer

File: 프로그래머스/2/42747. H－Index/H－Index.java
import java.util.*;

class Solution {
    public int solution(int[] citations) {
        int answer = 0;
        int length = citations.length;
        
        Arrays.sort(citations);
        
        for (int i = 0; i < length; i++) {
            if (citations[i] >= length - i) {
                answer = length - i;
                break;
            }
        }
        
        return answer;
    }
}